controlplane/internal/controller/openshiftassistedcontrolplane_controller.go

```diff
@@ -500,30 +508,73 @@ func (r *OpenshiftAssistedControlPlaneReconciler) ensureClusterDeployment(
 	acp *controlplanev1alpha2.OpenshiftAssistedControlPlane,
 	clusterName string,
 ) error {
-	if acp.Status.ClusterDeploymentRef == nil {
-		clusterDeployment := assistedinstaller.GetClusterDeploymentFromConfig(acp, clusterName)
-		_ = controllerutil.SetOwnerReference(acp, clusterDeployment, r.Scheme)
-		if err := util.CreateOrUpdate(ctx, r.Client, clusterDeployment); err != nil {
+	if acp.Status.ClusterDeploymentRef != nil {
+		return nil
+	}
```

Comment:
Wouldn't we still want to go through the create or patch function below even if the cluster deployment ref is set in the ACP's status? Like in the event the ACP's pull secret ref change?

Reply:
I agree generally, but this was the behavior before so I just kept it as is for this PR.

Reply:
Gotcha, sorry didn't catch that it was prior behavior.